apps/editing-toolkit/editing-toolkit-plugin/premium-content/blocks/subscriber-view/edit.js

```diff
@@ -3,53 +3,60 @@
  */
 import { InnerBlocks } from '@wordpress/block-editor';
 import { __ } from '@wordpress/i18n';
+import { useDispatch, withSelect } from '@wordpress/data';
 import { compose } from '@wordpress/compose';
-import { withSelect } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import Context from '../container/context';
 
-/**
- * Block edit function
- *
- * @typedef { object } Props
- * @property { string } clientId
- * @property { boolean } hasInnerBlocks
- *
- * @param { Props } props Properties
- */
-const Edit = ( props ) => (
-	<Context.Consumer>
-		{ ( { selectedTab, stripeNudge } ) => (
-			/** @see https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/no-static-element-interactions.md#case-the-event-handler-is-only-being-used-to-capture-bubbled-events */
-			// eslint-disable-next-line
-			<div hidden={ selectedTab.id === 'wall' } className={ selectedTab.className }>
-				{ stripeNudge }
-				<InnerBlocks
-					renderAppender={ ! props.hasInnerBlocks && InnerBlocks.ButtonBlockAppender }
-					templateLock={ false }
-					template={ [
-						[
-							'core/paragraph',
-							{
-								placeholder: __(
-									'Insert the piece of content you want your visitors to see after they subscribe.',
-									'full-site-editing'
-								),
-							},
-						],
-					] }
-				/>
-			</div>
-		) }
-	</Context.Consumer>
-);
+function Edit( { hasInnerBlocks, parentClientId, isSelected } ) {
+	const { selectBlock } = useDispatch( 'core/block-editor' );
+
+	useEffect( () => {
+		if ( isSelected ) {
+			selectBlock( parentClientId );
+		}
+	} );
```

Comment:
Same question as the logged out view, should we add a dependency array to this hook?